### PR TITLE
Fix leaks

### PR DESCRIPTION
**Pull request by pshopper**

### CODE REVIEW

File: src/server/responder.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+func NewStreamResponder(ctx context.Context, response chan<- *PushResult) *StreamResponder {
+	return &StreamResponder{
+		ctx:      ctx,
+		response: response,
+	}
+}
+
+func (s *StreamResponder) Send(projectId string, failures *DeviceIdList) {
```
Comment (Badmoonz) [inline src/server/responder.go]: добавить error

File: src/server/pushing_server.go [hunk 2]
```diff
@@ -27,51 +27,34 @@ func peerTypeProtobufToMPS(peerType PeerType) int {
...
 	}
 }
 
-func (p PushingServerImpl) Ping(ctx context.Context, ping *PingRequest) (*PongResponse, error) {
+func (p *PushingServerImpl) Ping(ctx context.Context, ping *PingRequest) (*PongResponse, error) {
 	return &PongResponse{}, nil
 }
 
-func streamOut(stream Pushing_PushStreamServer, responses chan *Response, errch chan error) {
+func streamOut(stream Pushing_PushStreamServer, responses chan *PushResult, errch chan error) {
```
Comment (Badmoonz) [inline src/server/pushing_server.go]: channel direction

File: src/server/pushing_server.go [hunk 3]
```diff
@@ -27,51 +27,34 @@ func peerTypeProtobufToMPS(peerType PeerType) int {
...
-	return resps
-}
-
-func (p PushingServerImpl) startStream(requests chan *Push, responses chan *Response) {
-	resps := p.getProvidersResponders()
-	for projectId, out := range resps {
-		// TODO: make timed output with aggregated results? [groupedWithin]
-		go workerOutputLoop(projectId, responses, out)
-	}
+func (p *PushingServerImpl) startStream(ctx context.Context, requests chan *Push, responses chan<- *PushResult) {
```
Comment (Badmoonz) [inline src/server/pushing_server.go]: channel directions

File: src/server/responder.go (hunk 1 above)
Comment (pshopper) [reply src/server/responder.go]: fixed

File: src/server/pushing_server.go (hunk 2 above)
Comment (pshopper) [reply src/server/pushing_server.go]: fixed

File: src/server/pushing_server.go (hunk 3 above)
Comment (pshopper) [reply src/server/pushing_server.go]: fixed

**Pull request merged**